src/models/replicate.js

```diff
@@ -0,0 +1,64 @@
+import Replicate from 'replicate';
+import { toSinglePrompt } from '../utils/text.js';
+
+// llama, mistral
+export class ReplicateAPI {
+	constructor(model_name, url) {
+		this.model_name = model_name;
+		this.url = url;
+
+		if (this.url) {
+			console.warn('Replicate API does not support custom URLs. Ignoring provided URL.');
+		}
+
+		if (!process.env.REPLICATE_API_KEY) {
+			throw new Error('Replicate API key missing! Make sure you set your REPLICATE_API_KEY environment variable.');
+		}
+
+		this.replicate = new Replicate({
+			auth: process.env.REPLICATE_API_KEY,
+		});
+	}
+
+	async sendRequest(turns, systemMessage) {
+		const stop_seq = '***';
+		let prompt_template;
+		const prompt = toSinglePrompt(turns, systemMessage, stop_seq);
+		let model_name = this.model_name || 'meta/meta-llama-3-70b-instruct';
+		if (model_name.includes('llama')) { // llama
+			prompt_template = "<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n\n<|eot_id|><|start_header_id|>user<|end_header_id|>\n\n{prompt}<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\n"
```

Comment:
Probably best not to leave system prompt blank. Why not use our system prompt instead of putting in in the message?

Reply:
you mean like in the parameters for the api call? some models support that but 70b instruct doesn't. anyway it's contained in the prompt itself.